### PR TITLE
Fix issues brought on PR

### PR DESCRIPTION
**Pull request by umamaistempo**

### CODE REVIEW

**Comment by umamaistempo** [comment]
Didn't go as far as i proposed on the referred PR but this is enough for now. I am too tired to continue work on this, so, at another moment i'll improve Software's code readability and improve overall type specifications

**Comment by renatomassaro** [comment]
uh, why not commit directly at the original branch/PR? I'll take a look

**Comment by umamaistempo** [comment]
Since those changes are mostly stylistic (except by very few bug fixes and the update on test suites) i'll use my collab status to merge those on your branch so they go to your PR.

Note that the current status of the HEAD includes stylistic problems according to credo but since most are related to websocket and you are refactoring websockets on https://github.com/HackerExperience/Helix/pull/229 , i'll just ignore them (and approve the merge on origin).

tl;dr, everything is fine now. I'll merge this into your branch, then into origin, then rebase your websocket PR on this HEAD.
